instrumentation/kubernetes-client-7.0/javaagent/src/test/groovy/KubernetesClientTest.groovy

```diff
@@ -18,51 +16,43 @@ import io.kubernetes.client.openapi.ApiException
 import io.kubernetes.client.openapi.apis.CoreV1Api
 import io.opentelemetry.instrumentation.test.AgentInstrumentationSpecification
 import io.opentelemetry.instrumentation.test.asserts.TraceAssert
-import io.opentelemetry.instrumentation.test.server.http.TestHttpServer
 import io.opentelemetry.semconv.trace.attributes.SemanticAttributes
+import io.opentelemetry.testing.armeria.common.HttpResponse
+import io.opentelemetry.testing.armeria.common.HttpStatus
+import io.opentelemetry.testing.armeria.common.MediaType
+import io.opentelemetry.testing.armeria.testing.junit5.server.mock.MockWebServerExtension
 import java.util.concurrent.CountDownLatch
-import java.util.concurrent.atomic.AtomicInteger
 import java.util.concurrent.atomic.AtomicReference
-import spock.lang.AutoCleanup
 import spock.lang.Shared
 
 class KubernetesClientTest extends AgentInstrumentationSpecification {
   private static final String TEST_USER_AGENT = "test-user-agent"
 
   @Shared
-  def responseStatus = new AtomicInteger()
-
-  @AutoCleanup
-  @Shared
-  TestHttpServer server
+  def server = new MockWebServerExtension()
 
   @Shared
   CoreV1Api api
 
-  def setup() {
-    // Lazy-load server to allow traits to initialize first.
-    if (server == null) {
-      server = httpServer {
-        handlers {
-          all {
-            handleDistributedRequest()
```

Comment:
I considered extracting the logic for this that is currently embedded in `HttpClientTest` to be able to use here. But it would be the one and only test needing this, so instead I just switched with an assertion on `traceparent` header which I guess tests the same thing